crates/compiler/parse/src/type_annotation.rs

```diff
@@ -588,7 +588,10 @@ fn record_type_field<'a>() -> impl Parser<'a, AssignedField<'a, TypeAnnotation<'
         // (This is true in both literals and types.)
         let (_, opt_loc_val, state) = optional(either(
             byte(b':', ETypeRecord::Colon),
-            byte(b'?', ETypeRecord::Optional),
+            and(
+                byte(b'?', ETypeRecord::Optional),
```

Comment:
Should probably now have two `ETypeRecord::Optional` error variants so our future parsing error messages can distinguish between the two

Reply:
Done